src/Command/ModelCommand.php

```diff
@@ -804,14 +811,14 @@ public function getValidation(Table $model, array $associations, Arguments $args
      * @param \Cake\Database\Schema\TableSchemaInterface $schema The table schema for the current field.
      * @param string $fieldName Name of field to be validated.
      * @param array $metaData metadata for field
-     * @param array<string> $primaryKey The primary key field. Unused because PK validation is skipped
+     * @param bool $isForeignKey Indicates if the currently precessed field is a foreign key
      * @return array Array of validation for the field.
      */
     public function fieldValidation(
         TableSchemaInterface $schema,
         string $fieldName,
         array $metaData,
-        array $primaryKey
+        bool $isForeignKey = false
```

Comment:
Won't this run into BC errors with userland extensions? Should we put the `isForeignKey` attribute into `metaData` instead?

Reply:
This is targeting `3.x`, so we can make breaking changes. But yes we should target `2.x` and put the `isForeignKey` in the `metaData`.

Reply:
So #841 should be fine in this regard